pysatCDAAC/instruments/cosmic2_ivm.py

```diff
@@ -0,0 +1,161 @@
+"""Supports IVM data from the COSMIC2 satellite.
+
+The Constellation Observing System for Meteorology, Ionosphere, and Climate
+(COSMIC) is comprised of six satellites in LEO with GPS receivers. The
+occultation of GPS signals by the atmosphere provides a measurement of
+atmospheric parameters. Data downloaded from the COSMIC Data Analaysis
+and Archival Center.
+
+More info about the dataset can be found at https://doi.org/10.5065/t353-c093
+
+Properties
+----------
+platform
+    'cosmic2'
+name
+    'ivm'
+tag
+    None supported
+inst_id
+    ['e1', 'e2', 'e3', 'e4', 'e5', 'e6']
+
+Warnings
+--------
+- Routine was not produced by COSMIC team
+
+"""
+
+import datetime as dt
+import functools
+import os
+import warnings
+
+import pysat
+from pysat.instruments.methods import general as mm_gen
+from pysatCDAAC.instruments.methods import general as mm_cdaac
+
+# ----------------------------------------------------------------------------
+# Instrument attributes
+
+platform = 'cosmic2'
+name = 'ivm'
+
+tags = {'': 'Ion Velocity Meter data'}
+
+inst_ids = {inst_id: list(tags.keys()) for inst_id in ['e1', 'e2', 'e3', 'e4',
+                                                       'e5', 'e6']}
+
+# Because all data products are stored in one tar file, inst_id not used
+directory_format = os.path.join('{platform}', '{name}', '{tag}')
+
+# ----------------------------------------------------------------------------
+# Instrument test attributes
+
+_test_dates = {inst_id: {'': dt.datetime(2021, 1, 1)}
+               for inst_id in inst_ids.keys()}
+
+# ----------------------------------------------------------------------------
+# Instrument methods
+
+
+def init(self):
+    """Initialize the Instrument object with instrument specific values.
+
+    Note
+    ----
+    Runs once upon instantiation.
+
+    """
+    ack = ' '.join(('FORMOSAT-7/COSMIC-2 is a partnership between the National',
+                    'Space Organization in Taiwan and NOAA, the U.S. Air',
+                    'Force, and the University Corporation for Atmospheric',
+                    'Research (UCAR) in the United States.'))
+    refs = ' '.join(('UCAR COSMIC Program, 2019: COSMIC-2 Data Products [Data',
+                     'set]. UCAR/NCAR - COSMIC, Access date [insert date],'
+                     'https://doi.org/10.5065/T353-C093'))
+    self.acknowledgements = ack
+    self.references = refs
+    pysat.logger.info(ack)
+    pysat.logger.info(' '.join(('All spacecraft data stored in single tar',
+                                'files on the CDAAC website. Pysat will',
+                                'download all and update accordingly.')))
+
+    return
+
+
+def clean(self):
+    """Clean COSMIC2 IVM data to the specified level.
+
+    Note
+    ----
+    'clean' - Not specified
+    'dusty' - Not specified
+    'dirty' - Not specified
+    'none'  No cleaning applied, routine not called in this case.
+
+    """
+    warnings.warn('Cleaning not yet implemented')
```

Comment:
What type of warning is this?

Reply:
Updated with a general version based on pysatNASA